code/src/java/pcgen/gui2/facade/Gui2InfoFactory.java

```diff
@@ -859,15 +859,15 @@ private HtmlInfoBuilder getEquipmentHtmlInfo(Equipment equip)
 
 		Integer a = EqToken.getMaxDexTokenInt(pc, equip);
 
-		if (a.intValue() != Constants.MAX_MAXDEX)
```

Comment:
should we replace it with Objects.equals() or AssertEquals()? IMHO relying on explicit boxing for this kind of subtle effect is asking for issues.